rts/Lua/LuaSyncedRead.cpp

```diff
@@ -8640,7 +8643,11 @@ int LuaSyncedRead::TraceRayGround(lua_State* L)
 	if (!ignoreWater)
 		traceLength = std::min(traceLength, waterRayLength);
 	lua_pushnumber(L, traceLength);
-	return 1;
+	auto result = pos + dir * traceLength;
```

Comment:
This won't be correct if `dir` was not normalized.

Reply:
Fixed.